samples/openapi3/server/petstore/spring-boot-oneof/src/main/java/org/openapitools/model/BarRef.java

```diff
@@ -0,0 +1,157 @@
+import org.openapitools.model.EntityRef;
+import org.openapitools.jackson.nullable.JsonNullable;
+import java.time.OffsetDateTime;
+import javax.validation.Valid;
+import javax.validation.constraints.*;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+
+import java.util.*;
+import javax.annotation.Generated;
+
+/**
+ * BarRef
+ */
+
+
+@Generated(value = "org.openapitools.codegen.languages.SpringCodegen")
+public class BarRef extends EntityRef implements BarRefOrValue {
+
+  @JsonProperty("href")
+  private String href;
+
+  @JsonProperty("@schemaLocation")
+  private String atSchemaLocation;
+
+  @JsonProperty("@baseType")
+  private String atBaseType;
+
+  @JsonProperty("@type")
+  private String atType;
+
+  public BarRef href(String href) {
+    this.href = href;
+    return this;
+  }
+
+  /**
+   * Hyperlink reference
+   * @return href
+  */
+  
+  @Schema(name = "href", description = "Hyperlink reference", required = false)
+  public String getHref() {
+    return href;
+  }
+
+  public void setHref(String href) {
+    this.href = href;
+  }
+
+  public BarRef atSchemaLocation(String atSchemaLocation) {
+    this.atSchemaLocation = atSchemaLocation;
+    return this;
+  }
+
+  /**
+   * A URI to a JSON-Schema file that defines additional attributes and relationships
+   * @return atSchemaLocation
+  */
+  
+  @Schema(name = "@schemaLocation", description = "A URI to a JSON-Schema file that defines additional attributes and relationships", required = false)
+  public String getAtSchemaLocation() {
+    return atSchemaLocation;
+  }
+
+  public void setAtSchemaLocation(String atSchemaLocation) {
+    this.atSchemaLocation = atSchemaLocation;
+  }
+
+  public BarRef atBaseType(String atBaseType) {
+    this.atBaseType = atBaseType;
+    return this;
+  }
+
+  /**
+   * When sub-classing, this defines the super-class
+   * @return atBaseType
+  */
+  
+  @Schema(name = "@baseType", description = "When sub-classing, this defines the super-class", required = false)
+  public String getAtBaseType() {
+    return atBaseType;
+  }
+
+  public void setAtBaseType(String atBaseType) {
+    this.atBaseType = atBaseType;
+  }
+
+  public BarRef atType(String atType) {
+    this.atType = atType;
+    return this;
+  }
+
+  /**
+   * When sub-classing, this defines the sub-class Extensible name
+   * @return atType
+  */
+  @NotNull 
+  @Schema(name = "@type", description = "When sub-classing, this defines the sub-class Extensible name", required = true)
+  public String getAtType() {
+    return atType;
+  }
+
+  public void setAtType(String atType) {
+    this.atType = atType;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(href, atSchemaLocation, atBaseType, atTypesuper.hashCode());
```

Comment:
The Bug caused by `org.openapitools.codegen.utils.OneOfImplementorAdditionalData#addToImplementor`